content/en/blog/istio-traffic-management-walkthrough/index.md

```diff
@@ -0,0 +1,1162 @@
+---
+title: "Hands-on walkthrough of traffic management in Istio"
+description: "In this blog, we will talk about how to get started with routing traffic between your services using service mesh. "
+date: "2021-03-03T12:15:00"
+author: "[Peter Jausovec](https://www.peterj.dev/)"
+# thumbnail
+thumbnail: "/images/blog/traffic.jpg"
+tags: ["istio"]
+---
+
+In this blog, I will talk about how to get started with routing traffic between your services using service mesh. You will learn how to set up an *Ingress* resource to allow traffic into your cluster as well as an *Egress* resource to enable traffic to exit your cluster. I will explain how to deploy a new version of the service and run it alongside with the released production version of the service without disrupting production traffic. With both service versions deployed we will gradually release the new version and start routing a percentage of incoming requests to the latest version. With this basic set up in place, we will show how to do more advanced with match conditions on request parameters (e.g. URL, headers). Finally, I will talk about mirroring production traffic to a newly deployed service without impacting end-users.
+
+## Deploying sample services
+
+In this blog, you will be using two different services. The first one, called Hello Web, is a simple frontend web application that makes calls to another service, called the Greeter Service. We will be using multiple versions of the Greeter Service to demonstrate the use of traffic routing. The diagram below shows the two services.
+
+![Hello web and greeter services](hello-greeter-arch.png)
+
+But, to get started, we will deploy the Hello Web and Greeter Service V1. Let's start with the Kubernetes deployment and service for the Greeter Service V1 and deploy the resources to the `default` namespace on your Kubernetes cluster.
+
+~~~~~~~~yaml
+cat <<EOF | kubectl create -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: greeter-service-v1
+  labels:
+    app: greeter-service
+    version: v1
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: greeter-service
+      version: v1
+  template:
+    metadata:
+      labels:
+        app: greeter-service
+        version: v1
+    spec:
+      containers:
+        - image: learnistio/greeter-service:1.0.0
```

Comment:
do we want to push these images to `tetratelabs` GCR?

Reply:
I think it can be kept on DockerHub, as GCR is not accessible in China.